src/transformers/training_args.py

```diff
@@ -507,20 +504,20 @@ def _setup_devices(self) -> Tuple["torch.device", int]:
             else:
                 torch.distributed.init_process_group(backend="nccl")
             device = torch.device("cuda", self.local_rank)
-            n_gpu = 1
+            self._n_gpu = 1
 
         if device.type == "cuda":
             torch.cuda.set_device(device)
 
-        return device, n_gpu
+        return device
```

Comment:
I'm curious, if it always returns a single device, why is it a plural in "setup_devices"?

Reply:
It's a private method in any case, so users are not supposed to call it at all. You call the `device` attribute if you want the deivce.